FoodBookApp/UI/Views/SpotDetail/SpotDetailView.swift

```diff
@@ -221,9 +221,13 @@ struct SpotDetailView: View {
         guard networkService.isOnline else { return }
         Task {
             do {
+                let startTime = Date()
                 try await model.fetchSpot(spotId: spotId)
                 try await model.fetchCategories()
                 loadingState = LoadingState.finished_loading
+                let finishTime = Date()
+                let time = (finishTime.timeIntervalSince(startTime) * 100).rounded() / 100
+                try await model.addFetchingTime(spotId: spotId, spotName: model.spot.name, time: time)
```

Comment:
I would try not to await this, just in the off change the the operation takes a long time for X/Y reason the user would still be left waiting for the view to load but the content they _actually_ need is already there. In the VM you wrap the function in a Task, so that here you don't have to await it. 

I did something similar with the bookmarks usage, you can see how [here](https://github.com/ISIS3510-202410-Team23/SwiftApp/blob/develop/FoodBookApp/Services/BookmarksService.swift?plain=1#L66) and [here](https://github.com/ISIS3510-202410-Team23/SwiftApp/blob/develop/FoodBookApp/Services/BookmarksService.swift?plain=1#L85)

Reply:
Okie, I'll merge the other two PRs and work on this later